### DISCUSSION TITLE
Is it possible to add the SSD1286 controller to tft_espi

### DISCUSSION THREAD

**Discussion by hannescam** [Bodmer/TFT_eSPI#2676] [New processor/display support or capability request]
I am wondering if it would be possible to add the SSD1286 (something like [LPU4CG031](https://datasheet.lcsc.com/lcsc/2303301800_ALPSALPINE-LPU4CG031B_C5439700.pdf)) controller to TFT_eSPI.

**Comment by Bodmer** [comment]
Yes it is possible, but since I do not have one of those displays, so it does not make sense for me to try and add it as any additions to the library must be tested and debugged.

**Reply by hannescam** [reply]
but would it be hard to do

**Reply by Bodmer** [reply]
Not hard but it can be time consuming tracking down problems if things do not work straight away.

**Reply by avtozapusk** [reply]
https://github.com/ZinggJM/SSD1283A/tree/master   Maybe it's worth taking this library and changing it a little?

For example, make small changes in this file  SSD1283A.cpp

Most likely you need to replace this piece of code:


void SSD1283A::init(void)
{
  digitalWrite(_cs, HIGH);
  SPI.begin();
  //SPI.beginTransaction( { SPI_SPEED, MSBFIRST, SPI_MODE0 } );
  if (_rst >= 0)
  {
    digitalWrite(_rst, LOW);
    delay(2);
    digitalWrite(_rst, HIGH);
  }
  setBackLight(true);
  delay(200);
  static const uint16_t SSD1283A_regValues[] PROGMEM =
  {
    0x10, 0x2F8E,
    0x11, 0x000C,
    0x07, 0x0021,
    0x28, 0x0006,
    0x28, 0x0005,
    0x27, 0x057F,
    0x29, 0x89A1,
    0x00, 0x0001,
    TFTLCD_DELAY16, 100,
    0x29, 0x80B0,
    TFTLCD_DELAY16, 30,
    0x29, 0xFFFE,
    0x07, 0x0223,
    TFTLCD_DELAY16, 30,
    0x07, 0x0233,
    0x01, 0x2183,
    0x03, 0x6830,
    0x2F, 0xFFFF,
    0x2C, 0x8000,
    0x27, 0x0570,
    0x02, 0x0300,
    0x0B, 0x580C,
    0x12, 0x0609,
    0x13, 0x3100,
  };
  _init_table16(SSD1283A_regValues, sizeof(SSD1283A_regValues));
  setRotation(_rotation);
  invertDisplay(false);
}
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


This piece will probably be correct

//////////////////////////////////////////////////////////


void SSD1283A::init(void)
{
  digitalWrite(_cs, HIGH);
  SPI.begin();
  //SPI.beginTransaction( { SPI_SPEED, MSBFIRST, SPI_MODE0 } );
  if (_rst >= 0)
  {
    digitalWrite(_rst, LOW);
    delay(2);
    digitalWrite(_rst, HIGH);
  }
  setBackLight(true);
  delay(200);
  static const uint16_t SSD1283A_regValues[] PROGMEM =
  {
   
/*
   0x10, 0x2F8E,
    0x11, 0x000C,
    0x07, 0x0021,
    0x28, 0x0006,
    0x28, 0x0005,
    0x27, 0x057F,
    0x29, 0x89A1,
    0x00, 0x0001,
    TFTLCD_DELAY16, 100,
    0x29, 0x80B0,
    TFTLCD_DELAY16, 30,
    0x29, 0xFFFE,
    0x07, 0x0223,
    TFTLCD_DELAY16, 30,
    0x07, 0x0233,
    0x01, 0x2183,
    0x03, 0x6830,
    0x2F, 0xFFFF,
    0x2C, 0x8000,
    0x27, 0x0570,
    0x02, 0x0300,
    0x0B, 0x580C,
    0x12, 0x0609,
    0x13, 0x3100,
  */
 
///////////////////////////////////////////////////////////////////////////////////////////
//                 This information is on page 27 of this document                       //
//////////https://datasheet.lcsc.com/lcsc/2303301800_ALPSALPINE-LPU4CG031B_C5439700.pdf///

 0x00, 0x0001,  // Oscillation start 00 0001
 0x10, 0x1F92,  // Power control (1) 10 1F92
 0x11, 0x0014,  // Power control (2) 11 0014
 0x28, 0x0006,  // Test Enable 28 0006
 0x00, 0x0001,  // Oscillation start 00 0001
 0x10, 0x1F92,  // Power control (1) 10 1F92
 0x11, 0x0014,  // Power control (2) 11 0014
 0x02, 0x0000,  // LCD drive AC control 02 0000
 0x12, 0x040B,  // Power control (3) 12 040B
 0x03, 0x4830,  // Entry mode 03 4830
 0x01, 0x31AF,  // Driver output control 01 31AF 
 0x07, 0x0033,  // Display control 07 0033
 0x42, 0xAF00,  // 1st Screen driving 42 AF00
 0x21, 0x0000,  // RAM address set 21 0000 
 0x44, 0x8300,  // Hor. RAM address 44 8300
 0x45, 0xAF00,  // Ver. RAM address 45 AF00
 0x2C, 0x3000,  // Oscillator frequency 2C 3000
 0x29, 0x0000,  // Analog Enable 29 0000
 0x13, 0x3000,  // Power Control 4 13 3000 


 
  };
  _init_table16(SSD1283A_regValues, sizeof(SSD1283A_regValues));
  setRotation(_rotation);
  invertDisplay(false);
}


/////////////////////////////////////////////////////////////////////////////////////////////////

I have such a display (SSD1286A SPI). 

If he's alive I can check it out

**Comment by hannescam** [comment]
maybe i will try
and will it require maintaning work or not
